src/main/kotlin/org/rust/lang/core/psi/visitors/RustVisitorEx.kt

```diff
@@ -1,10 +1,19 @@
 package org.rust.lang.core.psi.visitors
 
+import com.intellij.psi.PsiFile
 import org.rust.lang.core.psi.RustLiteral
 import org.rust.lang.core.psi.RustVisitor
+import org.rust.lang.core.psi.impl.RustFile
 
 open class RustVisitorEx : RustVisitor() {
     open fun visitLiteral(literal: RustLiteral) = visitElement(literal)
     open fun visitNumericLiteral(literal: RustLiteral.Number) = visitLiteral(literal)
     open fun visitTextLiteral(literal: RustLiteral.Text) = visitLiteral(literal)
+    open fun visitRustFile(file: RustFile) {
+        super.visitFile(file)
+    }
+
+    override fun visitFile(file: PsiFile) {
+        if (file is RustFile) visitRustFile(file) else super.visitFile(file)
+    }
```

Comment:
`visitRustFile` looks to be just a convenience method. It could be easily removed and that `if (file is RustFile) ...` pattern could be used instead. I don't see any disadvantages of this other than you have to be careful when override `visitFile`, which probably will never happen having in mind that this is `RustVisitorEx`